### PR TITLE
For atscppapi::AsyncTimer, pass the thread pool for execution to the constructor, …

### PR DESCRIPTION
**Pull request by ywkaras**
…rather than always using the default thread pool.

### CODE REVIEW

File: lib/cppapi/AsyncTimer.cc [hunk 1]
```diff
@@ -62,9 +67,9 @@ handleTimerEvent(TSCont cont, TSEvent event, void *edata)
 }
 } // namespace
 
-AsyncTimer::AsyncTimer(Type type, int period_in_ms, int initial_period_in_ms)
+AsyncTimer::AsyncTimer(Type type, int period_in_ms, int initial_period_in_ms, TSThreadPool thread_pool)
```
Comment (SolidWallOfCode) [inline lib/cppapi/AsyncTimer.cc]: If `thread_pool` had a default of the `TSThreadPool` value of `ET_NET` then existing plugins would compile and run as they do now without functional change.

Comment (SolidWallOfCode) [reply lib/cppapi/AsyncTimer.cc]: That is, `TS_THREAD_POOL_DEFAULT`.

Comment (ywkaras) [reply lib/cppapi/AsyncTimer.cc]: I accomplish this by having multiple constructors, see the header file changes below.

**Comment by zwoop** [comment]
@ywkaras @SolidWallOfCode Do you guys want this in for v8.0.0?

**Pull request merged**